src/dali_executor/utils/utils.h

```diff
@@ -76,6 +78,31 @@ TensorListShape<Dims> cat_list_shapes(const Container &shapes) {
   return result;
 }
 
+template<typename Container, int Dims = -1>
+std::vector<TensorListShape<Dims>> split_list_shape(const TensorListShape<Dims> &shape,
+                                                    const Container &batch_sizes) {
+  size_t nresults = 0;
+  int64_t nsamples = 0;
+  for (const auto &bs : batch_sizes) {
+    nsamples += bs;
+    nresults++;
```

Comment:
Do you expect the `Container` to support range iteration but no `size` method?

Reply:
changed to use .size()